### PR TITLE
Moving all the things

### PR DESCRIPTION
**Pull request by Sollace**
There we go. I have some other commits but I'll just make sure the new package structure works first.

@warriordog Oh, and I'm supposed to do this so you don't miss it.

### CODE REVIEW

**Comment by warriordog** [comment]
Sweet, I will get this when I am home.  Im on mobile on a camping trip right now

**Comment by warriordog** [comment]
@Sollace 

This looks good, but why are some of the interfaces renamed back to the "I" prefix?  There is no need for it.

**Comment by BigXplosion** [comment]
Maybe because he's used to it... I'm still doing that because I'm used to :D

**Comment by warriordog** [comment]
I did it for a while, before I realized its kind of a java standards
violation.

**Comment by Sollace** [comment]
@warriordog I use the I prefix because it makes it easier to tell what is a class and what is not. Just like enums usually have an E.

It's actually a standard naming convention used in other languages and I see no reason not to use it in Java as well.

**Comment by warriordog** [comment]
@sollace

There is nothing wrong with it, but it looks odd from an object oriented
perspective.

**Comment by Sollace** [comment]
@warriordog 
You might find it looks odd but I find it helps a lot with making the code more readable. Just a couple days ago I remember looking through LiteLoader's code to try and find where GameEngine.registerResourcePack went. Now Mumfry hasn't used the "I" convention at all so there was no way for me to tell what was a class and what was not at a glance. You pretty much have to find something that's not an interface, which was hella difficult, and work back from there.

Turns out more than half of his code is interfaces, and a lot of what's not is abstract. I eventually found what I had to do to fix the last compile error in BlazeLoader which I'll probably include after the package structures are sorted out but it would have been far less painless if I could see that they were interfaces from the filename.

Of course looking back, knowing where the answer is, and that the Type Hierarchy works on interfaces, I can see how I could have found the solution much faster.

tl;dr It makes it easier to read, and especially helps in languages where it's not clear whether you're extending a class or implementing an interface. You can actually see it mentioned here in the MSDN naming conventions for C#. [linky](https://msdn.microsoft.com/en-us/library/8bc1fexb%28v=vs.71%29.aspx)

**Comment by warriordog** [comment]
@Sollace 

You know with any decent IDE you can right click a field and select "open type", "goto type", or something along those lines.  And if you are in eclipse or IDEA the class icon will have an "I" for interface.  And if you want to know what implemented the interface then right click -> "find usages" -> "extends/implements".  But I will go ahead and merge this anyway.

**Comment by warriordog** [comment]
Ah crap I forgot I promised myself no more base-package refactors...  Well congratulations BL for reaching 5 complete entire-api repackages, before ever being released...

**Comment by BigXplosion** [comment]
it's better than while it is release though
